ext/LegendDataManagementLegendHDF5IOExt.jl

```diff
@@ -172,16 +177,22 @@ function LegendDataManagement.read_ldata(f::Base.Callable, data::LegendData, rse
     end
 end
 
-_skipnothingmissing(xv::AbstractVector) = [x for x in skipmissing(xv) if !isnothing(x)]
-lflatten(x) = fast_flatten(collect(_skipnothingmissing(x)))
-lflatten(nt::AbstractVector{<:NamedTuple}) = flatten_by_key(collect(_skipnothingmissing(nt)))
-
-function LegendDataManagement.read_ldata(f::Base.Callable, data::LegendData, rsel::Tuple{DataTierLike, AbstractVector{FileKey}, ChannelOrDetectorIdLike}; kwargs...)
-    if !isempty(string(rsel[3]))
-        lflatten([LegendDataManagement.read_ldata(f, data, (rsel[1], fk, rsel[3]); kwargs...) for fk in rsel[2]])
-    else
-        lflatten([LegendDataManagement.read_ldata(f, data, (rsel[1], fk); kwargs...) for fk in rsel[2]])
+function LegendDataManagement.read_ldata(f::Base.Callable, data::LegendData, rsel::Tuple{DataTierLike, AbstractVector{FileKey}, ChannelOrDetectorIdLike}; parallel::Bool=false, wpool::WorkerPool=default_worker_pool(), kwargs...)    
+    @everywhere begin
+        @assert isdefined(Main, :LegendDataManagement) "Parallel read requires LegendDataManagement.jl and LegendHDF5IO.jl to be loaded on each worker.g. via `@everywhere using LegendDataManagement LegendHDF5IO`"
+        @assert isdefined(Main, :LegendHDF5IO) "Parallel read requires LegendDataManagement.jl and LegendHDF5IO.jl to be loaded on each worker.g. via `@everywhere using LegendDataManagement LegendHDF5IO`"
```

Comment:
This can become one line:
```suggestion
        @assert isdefined(Main, :LegendDataManagement) && isdefined(Main, :LegendHDF5IO) "Parallel read requires LegendDataManagement.jl and LegendHDF5IO.jl to be loaded on each worker, e.g. via `@everywhere using LegendDataManagement LegendHDF5IO`"
```

Reply:
Done

Reply:
Done